src/linter/valid_examples.cc

```diff
@@ -73,26 +74,53 @@ auto ValidExamples::condition(
 
   Evaluator evaluator;
   std::size_t cursor{0};
+  this->invalid_indices_.clear();
+  std::ostringstream collected_messages;
+
   for (const auto &example : schema.at("examples").as_array()) {
     const std::string ref{"$ref"};
     SimpleOutput output{example, {std::cref(ref)}};
     const auto result{
         evaluator.validate(schema_template, example, std::ref(output))};
     if (!result) {
-      std::ostringstream message;
-      message << "Invalid example instance at index " << cursor << "\n";
-      output.stacktrace(message, "  ");
-      return message.str();
+      this->invalid_indices_.insert(cursor);
+      collected_messages << "Invalid example instance at index " << cursor
+                         << "\n";
+      output.stacktrace(collected_messages, "  ");
     }
 
     cursor += 1;
   }
 
+  if (!this->invalid_indices_.empty()) {
+    return collected_messages.str();
+  }
+
   return false;
 }
 
 auto ValidExamples::transform(sourcemeta::core::JSON &schema) const -> void {
-  schema.erase("examples");
+  if (!schema.is_object() || !schema.defines("examples") ||
```

Comment:
We shouldn't need any of these checks. The `condition` should not apply if these are not true?